### ISSUE TITLE
subscriptions key validation

### ISSUE THREAD

**Issue by Eth3rnit3** [simukappu/activity_notification#126]
Hello, first of all, I wanted to impregnate you for the work you've done. I just implemented this gem in a very big application, even if it took me 2 days, thanks to you, I gained a lot more!

I was able to easily implement slack as well as other notification thanks to base and everything works very well.

There's still something I don't understand or that I had to do wrong, but I don't understand why when a user save his subscription for a given key for example `message.reply` then no other user can subscribe to this key, and those because of this validation
https://github.com/simukappu/activity_notification/blob/3dc335d461636cf8b3d70e7d0d9fb58318ccd0a6/lib/activity_notification/orm/mongoid/subscription.rb#L31 
```shell
pry(main)> a.find_or_create_subscription('message.reply').subscribe
=> true

pry(main)> b.find_or_create_subscription('message.reply').subscribe
ActivityNotification::RecordInvalidError: Key est déjà pris.
from /Users/eth3rnit3/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/activity_notification-2.1.1/lib/activity_notification/models/concerns/subscriber.rb:53:in `create_subscription'
```

I think that each user can subscribe to the same key no?

**Comment by sebfie** [comment]
@simukappu I also have this issue !

**Comment by Eth3rnit3** [comment]
I can give you more information about the problem...
it's probably related to **mongoid** on this line https://github.com/simukappu/activity_notification/blob/master/lib/activity_notification/orm/mongoid.rb#L47 I think what's expected is the creation of an `@target` instance variable is the assignment of `self.target` to it

I don't know why, but with **mongoid** the result is `nil`.

I can solve my problem by replacing the `:target` of this line https://github.com/simukappu/activity_notification/blob/master/lib/activity_notification/orm/mongoid/subscription.rb#L31 by `:target_id`

**Comment by simukappu** [comment]
I've recognized this mongoid subscription bug from v2.1.0. Thank you for your feedback!

**Comment by Eth3rnit3** [comment]
gladly

**Comment by simukappu** [comment]
Confirmed and merged your pull request #128. Thank you!

**Comment by Eth3rnit3** [comment]
Thank you!

**Comment by simukappu** [comment]
Just released as v2.1.2. Thank you for your contribution!
